### ISSUE TITLE
Statistics interface always killed by updater regardless if actually updated.

### ISSUE THREAD

**Issue by shacknetisp** [red-eclipse/master#1]
The update script currently does not check if the statistics interface was actually updated by the `git pull --rebase` before restarting it. This would be easily solved by comparing the git commit hash before and after (from `git rev-parse HEAD` for example) and only restarting if they differ, but I don't know the proper way to push a change to both this repository and the running scripts.

**Comment by TheAssassin** [comment]
You might be able to do it with this nice peace of code I use in another project:

```
git fetch

if [ "$(git rev-parse HEAD)" != "$(git rev-parse origin/master)" ];
    git pull --rebase
    ... restart
fi
```

Question is whether you're still subject to a non-bash shell, the script is unlike to work then.

**Comment by qreeves** [comment]
This was changed to do this on purpose to stop the stats server from consuming all the memory on the server.

**Comment by shacknetisp** [comment]
I see, in that case it may be better to only restart when the current process is over an hour or so old. There is a bit of a startup delay, and with the frequent updates it is down for a noticeable portion of time.

**Comment by qreeves** [comment]
Hourly kills implemented in 3025b33
